servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -4767,13 +4799,30 @@ public Assertion putAssertion(ResourceContext ctx, String domainName, String pol
         }
 
         // validate to make sure we have expected values for assertion fields
+        // - and also to make sure that the associated role exists
 
         validatePolicyAssertion(assertion, caller);
 
         dbService.executePutAssertion(ctx, domainName, policyName, null, assertion, auditRef, caller);
         return assertion;
     }
 
+    private List<String> getRoleNameList(String domainName, String caller) {
```

Comment:
@mendi160 we don't really need this function. All this function can just be simply replaced by a single line:

dbService.listRoles(domainName)'

that gives all the roles for the domain without creating additional objects, etc.